### PR TITLE
Add three cases in sriov test

### PR DESCRIPTION
**Pull request by yafu-1**
Add two cases in sriov test:
1.Attach the VF of VF Pool to the guest with VFIO driver with "managed=no";
2.Define network with duplicate vf list;
3.Add alias name in the interface xml;

Signed-off-by: yafu <yafu@redhat.com>

### CODE REVIEW

**Comment by balamuruhans** [comment]
@santwana would you like to review this PR ?

File: libvirt/tests/src/sriov/sriov.py [hunk 1]
```diff
@@ -470,33 +478,39 @@ def attach_interface():
...
-            if max_vfs_attached:
-                interface_list = []
-                for vf_addr in vf_list:
-                    new_iface = create_interface()
-                    mac_addr = new_iface.mac_address
-                    nodedev_pci_addr = create_nodedev_pci(vf_addr)
-                    attach_interface()
-                    interface_list.append(new_iface)
+            if inactive_pool:
+                expected_error = "is not active"
```
Comment (chunfuwen) [inline libvirt/tests/src/sriov/sriov.py]: move it to cfg file

File: libvirt/tests/src/sriov/sriov.py [hunk 2]
```diff
@@ -470,33 +478,39 @@ def attach_interface():
...
+                nodedev_pci_addr = create_nodedev_pci(vf_addr)
+                origin_driver = os.readlink("%s/%s/driver" % (pci_device_dir, vf_addr)).split('/')[-1]
+                logging.log("The driver of vf before attaching to guest is %s\n", driver)
                 count = 0
-                for new_iface in interface_list:
-                    vf_addr = vf_list[count]
-                    vf_addr_attrs = new_iface.hostdev_address.attrs
+                while count < int(loop_times):
+                    interface = attach_interface()
+                    if vf_type == "vf" or vf_type == "vf_pool":
```
Comment (chunfuwen) [inline libvirt/tests/src/sriov/sriov.py]: maybe changed to if vf_type in ["vf", "vf_pool"]

File: libvirt/tests/src/sriov/sriov.py [hunk 3]
```diff
@@ -470,33 +478,39 @@ def attach_interface():
...
                     count += 1
+                if max_vfs_attached:
+                    interface_list = []
+                    for vf_addr in vf_list:
+                        new_iface = create_interface()
+                        mac_addr = new_iface.mac_address
+                        nodedev_pci_addr = create_nodedev_pci(vf_addr)
+                        attach_interface()
+                        interface_list.append(new_iface)
+                    count = 0
```
Comment (chunfuwen) [inline libvirt/tests/src/sriov/sriov.py]: I assume count is not referred after this line code

File: libvirt/tests/src/sriov/sriov.py [hunk 4]
```diff
@@ -470,33 +478,39 @@ def attach_interface():
...
+                    for new_iface in interface_list:
+                        vf_addr = vf_list[count]
+                        vf_addr_attrs = new_iface.hostdev_address.attrs
+                        detach_interface()
+                        count += 1
```
Comment (chunfuwen) [inline libvirt/tests/src/sriov/sriov.py]: the same with previous comment

File: libvirt/tests/src/sriov/sriov.py [hunk 5]
```diff
@@ -470,33 +478,39 @@ def attach_interface():
...
+                    interface = attach_interface()
+                    if vf_type == "vf" or vf_type == "vf_pool":
+                        vf_addr_attrs = interface.hostdev_address.attrs
+                    if operation != "":
+                        do_operation()
                     detach_interface()
                     count += 1
+                if max_vfs_attached:
+                    interface_list = []
+                    for vf_addr in vf_list:
```
Comment (chunfuwen) [inline libvirt/tests/src/sriov/sriov.py]: I assume 502 to line 513 can be merged into one loop

File: libvirt/tests/src/sriov/sriov.py [hunk 6]
```diff
@@ -507,6 +521,14 @@ def attach_interface():
             netxml = create_hostdev_network()
             result = virsh.net_define(netxml.xml, ignore_status=True, debug=True)
             utils_test.libvirt.check_result(result, expected_fails=expected_error)
+        if duplicate_vf:
+            vf_list[5] = vf_list[0]
+            expected_error = "can only be listed once in network"
```
Comment (chunfuwen) [inline libvirt/tests/src/sriov/sriov.py]: move it to cfg file

File: libvirt/tests/src/sriov/sriov.py [hunk 7]
```diff
@@ -507,6 +521,14 @@ def attach_interface():
...
+            netxml = create_hostdev_network()
+            result = virsh.net_define(netxml.xml, ignore_status=True, debug=True)
+            utils_test.libvirt.check_result(result, expected_fails=expected_error)
```
Comment (chunfuwen) [inline libvirt/tests/src/sriov/sriov.py]: normally we use libvirt_vm.check_exit_status to check result.

Comment (chunfuwen) [inline libvirt/tests/src/sriov/sriov.py]: normally we use libvirt.check_exit_status  to check result

File: libvirt/tests/src/sriov/sriov.py [hunk 8]
```diff
@@ -350,6 +356,7 @@ def attach_interface():
         """
         if managed == "no":
             result = virsh.nodedev_detach(nodedev_pci_addr)
+            logging.debug(nodedev_pci_addr)
```
[the remark this answers is not in the document]
Comment (yafu-1) [reply libvirt/tests/src/sriov/sriov.py]: Done.

File: libvirt/tests/src/sriov/sriov.py (hunk 1 above)
Comment (yafu-1) [reply libvirt/tests/src/sriov/sriov.py]: Done.

File: libvirt/tests/src/sriov/sriov.py (hunk 2 above)
Comment (yafu-1) [reply libvirt/tests/src/sriov/sriov.py]: Done.

File: libvirt/tests/src/sriov/sriov.py (hunk 6 above)
Comment (yafu-1) [reply libvirt/tests/src/sriov/sriov.py]: Done.

File: libvirt/tests/src/sriov/sriov.py (hunk 7 above)
Comment (yafu-1) [reply libvirt/tests/src/sriov/sriov.py]: Done.

File: libvirt/tests/src/sriov/sriov.py [hunk 9]
```diff
@@ -110,7 +115,8 @@ def _vf_init_completed():
 
         pci_list_sriov = set(pci_list_sriov)
         pci_diff = list(pci_list_sriov.difference(pci_list_before))
-        net_diff = utils_misc.wait_for(_vf_init_completed, timeout=120)
+        net_diff = utils_misc.wait_for(_vf_init_completed, timeout=60)
+        print (net_diff)
```
Comment (balamuruhans) [inline libvirt/tests/src/sriov/sriov.py]: logging.info() can be used or can be removed if used during dev debug.

File: libvirt/tests/src/sriov/sriov.py [hunk 10]
```diff
@@ -454,7 +463,7 @@ def attach_interface():
             vf_num = max_vfs
             create_vfs(vf_num)
         else:
-            vf_num = max_vfs/2 + 1
+            vf_num = int(max_vfs/2) + 1
```
Comment (balamuruhans) [inline libvirt/tests/src/sriov/sriov.py]: `max_vfs // 2` for compatibility.

File: libvirt/tests/src/sriov/sriov.py [hunk 11]
```diff
@@ -470,43 +479,54 @@ def attach_interface():
...
             vf_list = []
             pf_addr = pci_id
             vf_list.append(pf_addr)
-            expected_error = "is not an SR-IOV Virtual Function"
             netxml = create_hostdev_network()
             result = virsh.net_define(netxml.xml, ignore_status=True, debug=True)
-            utils_test.libvirt.check_result(result, expected_fails=expected_error)
+            utils_test.libvirt.check_exit_status(result, expected_error)
+        if duplicate_vf:
+            vf_list[5] = vf_list[0]
```
Comment (balamuruhans) [inline libvirt/tests/src/sriov/sriov.py]: instead of hardcoding index can it be appended ?

File: libvirt/tests/src/sriov/sriov.py [hunk 12]
```diff
@@ -470,43 +479,54 @@ def attach_interface():
...
-                    attach_interface()
-                    interface_list.append(new_iface)
+            if inactive_pool:
+                result = virsh.attach_device(vm_name, file_opt=new_iface.xml, flagstr=option,
+                                             ignore_status=True, debug=True)
+                utils_test.libvirt.check_exit_status(result, expected_error)
+            else:
+                mac_addr = new_iface.mac_address
+                nodedev_pci_addr = create_nodedev_pci(vf_addr)
+                origin_driver = os.readlink("%s/%s/driver" % (pci_device_dir, vf_addr)).split('/')[-1]
```
Comment (balamuruhans) [inline libvirt/tests/src/sriov/sriov.py]: `os.path.join(pci_device_dir, vf_addr, driver)`
here and in L334,

File: libvirt/tests/src/sriov/sriov.py (hunk 9 above)
Comment (yafu-1) [reply libvirt/tests/src/sriov/sriov.py]: Done.

File: libvirt/tests/src/sriov/sriov.py (hunk 10 above)
Comment (yafu-1) [reply libvirt/tests/src/sriov/sriov.py]: Done.

File: libvirt/tests/src/sriov/sriov.py (hunk 11 above)
Comment (yafu-1) [reply libvirt/tests/src/sriov/sriov.py]: Done.

File: libvirt/tests/src/sriov/sriov.py (hunk 12 above)
Comment (yafu-1) [reply libvirt/tests/src/sriov/sriov.py]: Done.

**Comment by balamuruhans** [comment]
@yafu-1 please rebase the PR.

**Pull request merged**